### PR TITLE
Run additional scripts in /usr/local/bin/start-notebook.d/

### PR DESCRIPTION
**Pull request by clkao**
Actual use cases can be:
- impose more restrictive ulimit
- cleanup user trash directory
- whitelist env variables (eg kubernetes service endpoint info removal)

### CODE REVIEW

File: base-notebook/start.sh [hunk 1]
```diff
@@ -111,6 +111,13 @@ else
     if [[ "$GRANT_SUDO" == "1" || "$GRANT_SUDO" == 'yes' ]]; then
         echo 'Container must be run as root to grant sudo permissions'
     fi
+    for f in /usr/local/bin/start-notebook.d/*; do
+      case "$f" in
+        *.sh)     echo "$0: running $f"; . "$f" ;;
+        *)        echo "$0: ignoring $f" ;;
```
Comment (minrk) [inline base-notebook/start.sh]: Shouldm't we run all executables in this directory, regardless of extension? Or is it explicitly required to be 'sourced' scripts?

Comment (clkao) [reply base-notebook/start.sh]: Right, for my use case calling `ulimit` it needs to be dot-sourced. I don't mind also running everything else if people think that's useful.

File: base-notebook/start.sh [hunk 2]
```diff
@@ -111,6 +111,13 @@ else
...
+      esac
+      echo
+    done
```
Comment (parente) [inline base-notebook/start.sh]: Do we want this logic to execute if the user runs the container with `--user root` too? As written, it only runs in the `else` block of the surrounding check for the container user ID.

Comment (clkao) [reply base-notebook/start.sh]: good point. I think this should be moved to the top out of the `if` doing uid check.  With that we can gradually refactor this giant if block into smaller scripts.

**Comment by minrk** [comment]
Nice!

**Pull request merged**